Source/Libraries/FaultData/DataWriters/Emails/EventEmailService.cs

```diff
@@ -146,6 +162,30 @@ private void SendEmail(EmailType email, Event evt, List<string> recipients, Date
             }
         }
 
+        public void SendScheduledEmail(ScheduledEmailType email, DateTime xdaNow, DateTime xdaPrev, DateTime xdaNext)
```

Comment:
If we're incorporating scheduled emails into the `EventEmailService`, then it's not really an `EventEmailService` anymore. Either it should be renamed, or we should consider splitting this into multiple classes.

I feel compelled to point out that the functionality in `EventEmailService` used to be split with another `EmailService`, but we merged the two classes as we began implementing email features for XDAv3. Perhaps somewhat ironically, that design was probably better suited to implementing scheduled emails. It might be worthwhile revisiting that original design for inspiration.

https://github.com/GridProtectionAlliance/openXDA/blob/v2.6/Source/Libraries/FaultData/DataWriters/EventEmailService.cs
https://github.com/GridProtectionAlliance/openXDA/blob/v2.6/Source/Libraries/FaultData/DataWriters/EmailService.cs

Reply:
I don't think these should be split. Beyond generating the contents of an email the Sending is pretty identical. If there are more significant deviations in the future, I would suggest splitting it into 3 parts: some base class, the scheduled email class, and the event email class.

Reply:
Let me elaborate. In v2.6, `EmailService.SendEmail()` was invoked by `EventEmailService.SendEmail()`. The idea would be to also have `ScheduledEmailService.SendEmail()` invoke the very same `EmailService.SendEmail()`. It was originally designed such that `EmailService` was a completely separate class and that `EventEmailService` contained a reference to it. We could implement `EmailService` as a base class, but then instances of `ScheduledEmailServices` wouldn't be able to share an instance with `EventEmailService`, for whatever that's worth.

I'm of the opinion that these two should eventually be split, if only for the fact that you would never use the `EventEmailService` instance to send scheduled emails or vice-versa. Even the collection of tables they use in the database was made completely separate. However, having thought on it for a bit, I do also think going back to the original design is a bit out of scope for this PR right now, so I'm fine if you just rename to `EmailService` at this point. I'll probably revisit this when fixing a GTC bug we found the other day that was introduced by the merger of those two v2.6 classes.